lib/iris/tests/unit/experimental/ugrid/test_meshcoord__dataview.py

```diff
@@ -0,0 +1,440 @@
+    def fetch_face_xs(keys):
+        # Get lazy points of the face-x coord in this cube.
+        lazy_points = cube.coord(face_x_coordname).core_points()
+        return lazy_points[keys]
+
+    def fetch_node_xs(keys):
+        # Get lazy points of the face-x coord from the cube.
+        # NOTE: to fit this into a "normal" cube, the cube needs a 'nodes'
+        # dimension, to which this will be attached.
+        lazy_points = cube.coord(node_x_coordname).core_points()
+        return lazy_points[keys]
+
+    def fetch_face_node_inds(keys):
+        # Get lazy indices from the face-nodes-connectivity coord in the cube.
+        # NOTE: to fit the indices into a "normal" cube, we store them as the
+        # *bounds* of a face-nodes-connectivity coord, which is mapped to the
+        # 'faces' dimension.
+        lazy_indices = cube.coord(face_node_inds_coordname).core_bounds()
+        return lazy_indices[keys]
+
+    # Work out what the actual shapes and dtypes of our data arrays will be.
+    co = cube.coord(face_x_coordname)
+    face_x_shape, face_x_dtype = co.shape, co.dtype
+    co = cube.coord(node_x_coordname)
+    node_x_shape, node_x_dtype = co.shape, co.dtype
+    co = cube.coord(face_node_inds_coordname)
+    facenodeinds_shape, facenodeinds_dtype = co.core_bounds().shape, co.dtype
+
+    # Do some quick sanity checks.
+    assert len(face_x_shape) == 1
+    assert len(node_x_shape) == 1
+    assert len(facenodeinds_shape) == 2
+    assert facenodeinds_shape == (face_x_shape[0], 4)
+    assert np.dtype(facenodeinds_dtype).kind == "i"
+
+    # Wrap the basic access operations as deferred-access arrays, using the
+    # ArrayMimic (func --> arraylike) and Dask (arraylike --> da.Array).
+    face_x_arraylike = ArrayMimic(
+        fetch_face_xs, shape=face_x_shape, dtype=face_x_dtype
+    )
+    lazy_face_x = da.from_array(face_x_arraylike, chunks="auto")
+    node_x_arraylike = ArrayMimic(
+        fetch_node_xs, shape=node_x_shape, dtype=node_x_dtype
+    )
+    lazy_node_x = da.from_array(node_x_arraylike, chunks="auto")
+    facenodeinds_arraylike = ArrayMimic(
+        fetch_face_node_inds,
+        dtype=facenodeinds_dtype,
+        shape=facenodeinds_shape,
+    )
+    lazy_facenodeinds = da.from_array(facenodeinds_arraylike, chunks="auto")
+
+    # Construct the appropriate lazy bounds calculation for the MeshCoord.
+    # This type of multidimensional indexing is not supported...
+    #    "lazy_face_bounds_x = lazy_node_x[lazy_facenodeinds]"
+    # ...so we build that calculation by iterating over the final (bounds)
+    # index, and stacking those to reconstruct the output.
+    # Within that operation, we also apply our 'array_index_with_missing'
+    # function, which ensures that any -1 indices yield NaN points.
+    lazy_face_bounds_x = da.stack(
+        [
+            array_index_with_missing(
+                lazy_node_x, lazy_facenodeinds[:, i_bound]
+            )
+            for i_bound in range(4)
```

Comment:
Range should vary with face index shape.

Reply:
Good spot.
Will do this